### PR TITLE
[tests-only] added tests for  multiuser share

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds test for sharing a file with multiple users(3-users)

### Related Issue
https://github.com/owncloud/client/issues/8971#issuecomment-909885070

### CODE REVIEW

File: test/gui/shared/steps/steps.py [hunk 1]
```diff
@@ -239,6 +239,20 @@ def step(context, receiver, resource, permissions):
     openSharingDialog(context, resource)
     shareItem = SharingDialog()
     shareItem.addCollaborator(receiver, permissions)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:242]: why did you move it from Page object to the steps?

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -239,6 +239,20 @@ def step(context, receiver, resource, permissions):
...
+def step(context, resource):
+    openSharingDialog(context, resource)
+    shareItem = SharingDialog()
+
+    for row in context.table[1:]:
+        receiver = row[0]
+        permissions = row[1]
+        shareItem.addCollaborator(receiver, permissions)
+
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:255]: since it is being used in many places, you should create a function, and try to reuse it.

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -248,6 +262,7 @@ def step(context, receiver, resource, permissions):
     openSharingDialog(context, resource)
     shareItem = SharingDialog()
     shareItem.addCollaborator(receiver, permissions, True)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:265]: same here, repeated code

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -281,6 +316,8 @@ def collaboratorShouldBeListed(context, receiver, resource, permissions):
         waitForObjectExists(names.scrollArea_permissionShare_QCheckBox).checked,
         ('share' in permissionsList),
     )
+    shareItem = SharingDialog()
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:320]: and here

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:242]: Because we only need this in few place not for all `addCollaborator`. So I thought it would be better to have it in here.

File: test/gui/shared/steps/steps.py (hunk 2 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:255]: The reason for doing this is that we already have `SharingDialog` object so if we make a new function for this then every time we will have two object of `SharingDialog` instead fo having to do that I did this.

File: test/gui/shared/steps/steps.py [hunk 5]
```diff
@@ -239,6 +239,20 @@ def step(context, receiver, resource, permissions):
...
     shareItem.addCollaborator(receiver, permissions)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
+
+
+@When('the user adds following collaborators of resource "|any|" using the client-UI')
+def step(context, resource):
+    openSharingDialog(context, resource)
+    shareItem = SharingDialog()
+
+    for row in context.table[1:]:
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:250]: Add a little bit of comment here

File: test/gui/shared/steps/steps.py [hunk 6]
```diff
@@ -1104,3 +1141,18 @@ def step(context, resource):
...
     publicLinkDialog = PublicLinkDialog()
     publicLinkDialog.changePassword(publicLinkName, password)
+
+
+@Then(
+    'the following users should be listed in as collaborators for file "|any|" on the client-UI'
+)
+def step(context, resource):
+    receiverCount = 0
+    for row in context.table[1:]:
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:1151]: Add a little bit of comment here as well

File: test/gui/shared/steps/steps.py [hunk 7]
```diff
@@ -239,6 +239,21 @@ def step(context, receiver, resource, permissions):
...
     shareItem.addCollaborator(receiver, permissions)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
+
+
+@When('the user adds following collaborators of resource "|any|" using the client-UI')
+def step(context, resource):
+    openSharingDialog(context, resource)
+    shareItem = SharingDialog()
+
+    #     In the following loop we are trying to share resource with given permission to one user at a time given from the data table in the feature file
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:250]: extra spaces
```suggestion
    # In the following loop we are trying to share resource with given permission to one user at a time given from the data table in the feature file
```

File: test/gui/shared/steps/steps.py [hunk 8]
```diff
@@ -239,6 +239,21 @@ def step(context, receiver, resource, permissions):
     openSharingDialog(context, resource)
     shareItem = SharingDialog()
     shareItem.addCollaborator(receiver, permissions)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:242]: we can create a function `closeSharingDialog`  in pageObject and call it here and other places when needed.

File: test/gui/shared/steps/steps.py (hunk 7 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:250]: This change was made by black

File: test/gui/shared/steps/steps.py [hunk 9]
```diff
@@ -265,14 +281,39 @@ def step(context, receiver, resource, permissions):
...
+            "name": "sharedWith",
+            "type": "QLabel",
+            "visible": 1,
+        }
     )
+
+    #     we use sharedWithObj list from above while verifying if users are listed or not.
+    #     For this we need an index value i.e receiverCount
+    #     For 1st user in the list the index will be 0 which is receiverCount default value
+    #     For 2nd user in the list the index will be 1 and so on
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:314]: not a big deal but could you please fromat them?
```suggestion
    # we use sharedWithObj list from above while verifying if users are listed or not.
    # For this we need an index value i.e receiverCount
    # For 1st user in the list the index will be 0 which is receiverCount default value
    # For 2nd user in the list the index will be 1 and so on
```

File: test/gui/shared/steps/steps.py [hunk 10]
```diff
@@ -1104,3 +1147,20 @@ def step(context, resource):
...
     publicLinkDialog = PublicLinkDialog()
     publicLinkDialog.changePassword(publicLinkName, password)
+
+
+@Then(
+    'the following users should be listed in as collaborators for file "|any|" on the client-UI'
+)
+def step(context, resource):
+    #     Here we are trying to verify if the user added in when step are listed in the client-UI or not
+    #     We now have a variable name receiverCount which is used in collaboratorShouldBeListed function call
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:1157]: same as above
```suggestion
    # Here we are trying to verify if the user added in when step are listed in the client-UI or not
    # We now have a variable name receiverCount which is used in collaboratorShouldBeListed function call
```

File: test/gui/shared/steps/steps.py (hunk 9 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:314]: @saw-jan This is also done by black

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by Talank — APPROVED** [review]
